sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/AvoidExcessiveInheritance.cs

```diff
@@ -103,6 +106,15 @@ protected override void Initialize(ParameterLoadingAnalysisContext context)
                 }, SyntaxKind.ClassDeclaration);
         }
 
+        private static string GetRootNamespace(INamedTypeSymbol namedSymbol)
+        {
+            return namedSymbol
+                        .ContainingNamespace
+                        .ToDisplayString()
+                        .SplitCamelCaseToWords()
+                        .First();

```

Comment:
This might crash if the class is not defined into a namespace (see Foddy for example).